src/types/KeycloakUserId.ts

```diff
@@ -0,0 +1,41 @@
+import { Newtype, iso as createNewtypeIsomorphism } from 'newtype-ts';
+import { ajv } from '../ajv';
+import { InternalValidationError } from '../errors';
+import type { JSONSchemaType } from 'ajv';
+
+interface KeycloakUserId
+	extends Newtype<{ readonly KeycloakUserId: unique symbol }, string> {}
+
+// The cast at the end of this definition invalidates ajv's magic checks to ensure that a schema aligns with a type definition.
+// This is because Newtype removes the type context of the actual data.
+// This just means we have to be VERY careful to ensure that our schema definition does
+// adhere to our type definition / expectations.
+const keycloakUserIdSchema = {
+	type: 'string',
+	format: 'uuid',
+} as unknown as JSONSchemaType<KeycloakUserId>;
+
+const isKeycloakUserId = ajv.compile(keycloakUserIdSchema);
+
+const stringToKeycloakUserId = (s: string): KeycloakUserId => {
+	if (!isKeycloakUserId(s)) {
+		throw new InternalValidationError(
+			'Invalid KeycloakUserId',
+			isKeycloakUserId.errors ?? [],
+		);
+	}
+	return s;
```

Comment:
Should this be a call to `wrap`? I'm not sure I understand the tradeoff here - maybe it's better as-is! Just trying to understand.

```suggestion
	return keycloakUserIdIsomorphism.wrap(s);
```

Reply:
Good catch yes it should

Reply:
I was looking at this and saying "oh dear, shouldn't our type checker have caught this" and realized that `isKeycloakUserId` is lying -- it is saying it is a UUID, not that it is the newtype.  I'm going to change the schema and the `isFoo` to be `uuidStringSchema` and `isUuidString` to prevent this kind of confusion.

Reply:
As a very belated followup here, after our conversation about how newtype works / how it does not actually modify data I'm not sure that `wrap` is actually any more meaningful than a type guard.  All wrap is doing is changing the type signature of the value (under the hood it is still a string).

To that end, this actually *is* fine, technically.

That being said, I think that it is cleaner and more explicit to devs to use wrap and `isUuid`.

I am, however, currently towards having the `User.keycloakUserId` attribute typed as a `KeycloakUserId` rather than a `Uuid` that would need to be explicitly wrapped before it could be used.  Less boilerplate for the same outcome.

Reply:
All righty -- I think the changes are small enough that this doesn't need another review (famous last words!).  I do think it's cleaner now (and we can even benefit from TS to at least nudge any devs writing tests into having proper static strings)